lib/cocina/models/validators/map_coordinates_validator.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'geo/coord'
+
+module Cocina
+  module Models
+    module Validators
+      # Validates that Purl matches the external identifier (druid)
+      class MapCoordinatesValidator
+        def self.validate(clazz, attributes)
+          new(clazz, attributes).validate
+        end
+
+        def initialize(clazz, attributes)
+          @clazz = clazz
+          @attributes = attributes
+        end
+
+        def validate
+          return unless meets_preconditions?
+
+          # valid condition
+          return if invalid_coordinates.empty?
+
+          raise ValidationError, "Invalid map coordinates for #{druid}."
+        end
+
+        private
+
+        attr_reader :clazz, :attributes
+
+        def meets_preconditions?
+          map_coordinates
+        end
```

Comment:
is this a standardized thing in validators? it's kind of an ambiguously named function, so if it's not required, i think it'd be better to just inline it as `return unless map_coordinates.any?`or something in `#validate`. but if it's part of the validator API that's fine